torch/csrc/utils/tensor_new.cpp

```diff
@@ -126,10 +134,17 @@ Tensor new_with_tensor(c10::DispatchKey dispatch_key, at::ScalarType scalar_type
   return other.slice();
 }
 
+
 std::vector<int64_t> compute_sizes(PyObject* seq) {
   std::vector<int64_t> sizes;
   THPObjectPtr handle;
   while (PySequence_Check(seq)) {
+    if (PyArray_Check(seq) && PyArray_NDIM((PyArrayObject*) seq) == 0) { 
```

Comment:
Let's also add an explicit test for [a] and [a, b] and [a, c] where a and b are scalar arrays and c is not.

Reply:
Having a mixture of scalar and non scalar arrays in the list will throw an exception - this logic was always there (in the ```recursive_store```). However, having only scalar arrays is perfectly fine.